demo/api/src/importer/pipes/parsers/data-transformer.pipe.ts

```diff
@@ -0,0 +1,209 @@
+import { LoggerService } from "@nestjs/common";
+import { ImportFieldMetadata } from "@src/importer/decorators/csv-column.decorator";
+import { FieldTransformerData } from "@src/importer/decorators/field-transformer.decorator";
+import { ImporterEntityClass } from "@src/importer/entities/base-target.entity";
+import {
+    columnIsBoolean,
+    columnIsDate,
+    columnIsFloat,
+    columnIsInteger,
+    isNullable,
+    TargetEntityProperties,
+    TargetEntityProperty,
+} from "@src/importer/entities/target-entity.utils";
+import { plainToInstance } from "class-transformer";
+import { validate } from "class-validator";
+import { parse, parseISO } from "date-fns";
+import jp from "jsonpath";
+import { Transform, TransformCallback } from "stream";
+
+import { ImporterPipe, PipeData, ValidationError } from "../importer-pipe.type";
+
+type ValidateAndTransformValueProperties = {
+    value: string;
+    propertyMetadata: TargetEntityProperty;
+    field: ImportFieldMetadata;
+};
+
+type DataTransformOptions = {
+    fields: ImportFieldMetadata[];
+    targetEntityProperties: TargetEntityProperties;
+    entityDateFormatString?: string;
+    fieldTransformers?: FieldTransformerData[];
+};
+type PipeDataAndErrors = {
+    data: PipeData;
+    errors: ValidationError[];
+};
+
+type PossibleValueTypes = string | number | boolean | Date | undefined | null | object;
+
+export class DataTransformerPipe implements ImporterPipe {
+    private readonly options: DataTransformOptions;
+
+    constructor(options: DataTransformOptions, private readonly entity: ImporterEntityClass) {
+        this.options = options;
+    }
+
+    getPipe(runLogger: LoggerService) {
+        return new DataTransformer(this.options, runLogger, this.entity);
+    }
+}
+
+type ParserPipeData = Record<string, string>;
+
+export class DataTransformer extends Transform {
```

Comment:
A discussion point for today's meeting: What are we doing better here than class-validator + class-transformer?